src/military/military.cpp

```diff
@@ -6437,6 +6431,29 @@ void update_siege_progress(sys::state& state) {
 			// is controler != owner ...
 			// event::fire_fixed_event(state, );
 		}
+		if(auto nr = new_rebel_controller.get(prov); nr) {
+			province::set_province_controller(state, prov, nr);
+			eject_ships(state, prov);
+
+			for(auto ar : state.world.province_get_army_location(prov)) {
+				auto a = ar.get_army();
```

Comment:
so, this is going to loop over all the armies in the province that has just been sieged ... presumably there are only rebel armies there, so none of them will be rebel hunters, which are exclusively player nation controlled...